tests/integration/base.py

```diff
@@ -0,0 +1,327 @@
+import ipaddress
+import json
+import os
+import tempfile
+from abc import abstractmethod
+
+from pathlib import Path
+from subprocess import Popen, PIPE, TimeoutExpired
+from logging import getLogger, StreamHandler, INFO
+from typing import Dict, Any, Optional, List, NamedTuple
+from unittest import TestCase
+
+import pytest
+
+from tests import REPO_ROOT
+
+PROJECT_NAME = "project"
+
+
+class CommandResult(NamedTuple):
+    process: Popen
+    stdout: str
+    stderr: str
+
+
+TIMEOUT = 300
+
+LOG = getLogger(__name__)
+LOG.addHandler(StreamHandler())
+LOG.setLevel(INFO)
+
+
+def run_command(command_list, cwd=None, env=None, timeout=TIMEOUT) -> CommandResult:
+    if not env:
+        env = os.environ.copy()
+    LOG.info("PATH=%s", env["PATH"])
+    process_execute = Popen(command_list, cwd=cwd, env=env, stdout=PIPE, stderr=PIPE)
+    try:
+        stdout_data, stderr_data = process_execute.communicate(timeout=timeout)
+        stdout = stdout_data.decode()
+        stderr = stderr_data.decode()
+        LOG.info(f"Stdout: {stdout}")
+        LOG.info(f"Stderr: {stderr}")
+        return CommandResult(process_execute, stdout, stderr)
+    except TimeoutExpired:
+        LOG.error(f"Command: {command_list}, TIMED OUT")
+        LOG.error(f"Return Code: {process_execute.returncode}")
+        process_execute.kill()
+        raise
+
+
+class Base:
```

Comment:
I am usually okay with less documentation in tests. But these parent classes are useful. Please add some documentation around these classes and methods.

Reply:
Added, good idea!